addons/mail/static/src/new/core/failure_model.js

```diff
@@ -0,0 +1,63 @@
+/** @odoo-module */
+
+import { _t } from "@web/core/l10n/translation";
+import { createLocalId } from "./thread_model.create_local_id";
+
+export class Failure {
+    /** @type {import("@mail/new/core/notification_model").Notification[]} */
+    notifications;
+    /** @type {string} */
+    modelName;
+    /** @type {string} */
+    resModel;
+    /** @type {number} */
+    lastMessageId;
+    /** @type {'sms' | 'email'} */
+    type;
+
+    static insert(state, data) {
+        const { modelName, type } = data;
+        let failure = state.menu.failures.find(
+            ({ localId }) => localId === createLocalId(modelName, type)
+        );
+        if (!failure) {
+            failure = new Failure(state, data);
+        }
+        failure.update(data);
+        return state.menu.failures.find(({ localId }) => failure.localId === localId);
+    }
+
+    constructor(state, data) {
+        const { modelName, type } = data;
+        this.localId = createLocalId(modelName, type);
+        this._state = state;
+        this._state.menu.failures.push(this);
+    }
+
+    update(data) {
+        this.notifications ??= data.notifications;
+        this.modelName ??= data.modelName;
+        this.resModel ??= data.resModel;
+        this.type ??= data.type;
```

Comment:
This pattern looks good, but it's wrong: here if `this.value` is non-nullish, `data` cannot change it.